core/domain/stats_services_test.py

```diff
@@ -115,6 +114,138 @@ def test_resolve_answers_for_default_rule(self):
+        self._record_answer('a', exploration_id='eid1', state_name='Second')
+        self._record_answer('b', exploration_id='eid1', state_name='Second')
+        self._record_answer('b', exploration_id='eid1', state_name='Second')
+        self._record_answer('b', exploration_id='eid1', state_name='Second')
+        self._record_answer('c', exploration_id='eid1', state_name='Second')
+        self._record_answer('c', exploration_id='eid0', state_name='First')
+
+        top_answers_dict = stats_services.get_top_state_rule_answers_multi([{
+            'exploration_id': 'eid0',
+            'state_name': 'First',
+            'rule_str_list': [self.DEFAULT_RULESPEC_STR]
+        }, {
+            'exploration_id': 'eid1',
+            'state_name': 'Second',
+            'rule_str_list': [self.DEFAULT_RULESPEC_STR]
+        }], 10)
+        self.assertEquals(len(top_answers_dict), 2)
+        self.assertEquals(top_answers_dict['eid0'], [{
+            'value': 'a',
+            'count': 1
+        }, {
+            'value': 'c',
+            'count': 1
+        }])
+        self.assertEquals(top_answers_dict['eid1'], [{
+            'value': 'b',
+            'count': 3
+        }, {
+            'value': 'a',
+            'count': 1
+        }, {
+            'value': 'c',
+            'count': 1
+        }])
+
+    def test_get_top_state_rule_answers_fixed_number_from_multiple_exps(self):
+        # There are no initial top answers for these explorations.
+        top_answers_dict = stats_services.get_top_state_rule_answers_multi([{
+            'exploration_id': 'eid0',
+            'state_name': 'First',
+            'rule_str_list': [self.DEFAULT_RULESPEC_STR]
+        }, {
+            'exploration_id': 'eid1',
+            'state_name': 'Second',
+            'rule_str_list': [self.DEFAULT_RULESPEC_STR]
+        }], 2)
+        self.assertEquals(len(top_answers_dict), 2)
+        self.assertEquals(len(top_answers_dict['eid0']), 0)
+        self.assertEquals(len(top_answers_dict['eid1']), 0)
+
+        # Submit some answers.
+        self._record_answer('a', exploration_id='eid0', state_name='First')
+        self._record_answer('b', exploration_id='eid1', state_name='Second')
+        self._record_answer('c', exploration_id='eid1', state_name='Second')
+        self._record_answer('d', exploration_id='eid1', state_name='Second')
+
+        # Only 2 answers were requested.
+        top_answers_dict = stats_services.get_top_state_rule_answers_multi([{
+            'exploration_id': 'eid0',
+            'state_name': 'First',
+            'rule_str_list': [self.DEFAULT_RULESPEC_STR]
+        }, {
+            'exploration_id': 'eid1',
+            'state_name': 'Second',
+            'rule_str_list': [self.DEFAULT_RULESPEC_STR]
+        }], 2)
+        self.assertEquals(len(top_answers_dict), 2)
+        self.assertEquals(len(top_answers_dict['eid0']), 1)
+        self.assertEquals(len(top_answers_dict['eid1']), 2)
+
```

Comment:
I don't think so to the first (since after the latest changes there's no win over testing multiple explorations compared to state names). The second sounds reasonable, though, so added! (I interpreted your second suggestion to mean multiple rule spec strings).